extensions/positron-r/src/r-installation.ts

```diff
@@ -128,10 +213,16 @@ export class RInstallation {
 			this.arch = '';
 		}
 
-		this.valid = true;
-
 		LOGGER.info(`R installation discovered: ${JSON.stringify(this, null, 2)}`);
 	}
+
+	toJSON() {
+		return {
+			...this,
+			reasonDiscovered: this.reasonDiscovered?.map(friendlyReason) ?? null,
+			reasonRejected: this.reasonRejected ? friendlyReason(this.reasonRejected) : null
+		};
```

Comment:
This makes human-readable reasons appear in the log. Here's an example of an unusable R installation:

```
2024-12-06 12:13:40.397 [info] R installation discovered: {
  "usable": false,
  "supported": true,
  "reasonDiscovered": [
    "Found in the primary location for R versions on this operating system"
  ],
  "reasonRejected": "Non-orthogonal installation that is also not the current version",
  "binpath": "/Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/bin/R",
  "homepath": "/Library/Frameworks/R.framework/Resources",
  "semVersion": {
    "options": {},
    "loose": false,
    "includePrerelease": false,
    "raw": "4.4.2",
    "major": 4,
    "minor": 4,
    "patch": 2,
    "prerelease": [],
    "build": [],
    "version": "4.4.2"
  },
  "version": "4.4.2",
  "arch": "arm64",
  "current": false,
  "orthogonal": false
}
```